operators/xr/README.md

```diff
@@ -0,0 +1,275 @@
+    def compose(self):
+        # Create XR session
+        xr_session = XrSession(
+            self,
+            application_name="My XR App",
+            application_version=1
+        )
+        
+        # Create composition layer manager
+        composition_manager = XrCompositionLayerManager(
+            self, xr_session=xr_session
+        )
+        
+        # Frame management operators
+        begin_frame = XrBeginFrameOp(
+            self, xr_session=xr_session
+        )
+        
+        end_frame = XrEndFrameOp(
+            self, xr_session=xr_session
+        )
+        
+        # Define workflow
+        self.add_flow(begin_frame, end_frame)
+```
+
+## Python API
+
+### Core Classes
+
+**XrSession**
+
+```python
+# Create XR session
+session = XrSession(
+    fragment,
+    application_name="My App",
+    application_version=1
+)
+
+# Access session properties
+view_configs = session.view_configurations()
+depth_range = session.view_configuration_depth_range()
+```
+
+**XrSwapchainCuda**
+
+```python
+# Create swapchain
+swapchain = XrSwapchainCuda(
+    session,
+    format=XrSwapchainCudaFormat.R8G8B8A8_SRGB,
+    width=1920, height=1080
+)
+
+# Acquire image for rendering
+tensor = swapchain.acquire()
+# ... render to tensor ...
+swapchain.release(cuda_stream)
+```
+
+**XrHandTracker**
+
+```python
+# Create hand tracker
+hand_tracker = XrHandTracker(
+    fragment,
+    xr_session=session,
+    hand=XrHandEXT.XR_HAND_LEFT_EXT
+)
+
+# Get hand joint locations
+joints = hand_tracker.locate_hand_joints()
+```
+
+### Data Types
+
+**Pose and Transform**
+
+```python
+# 3D position
+position = XrVector3f(x=0.0, y=0.0, z=0.0)
+
+# Quaternion rotation
+orientation = XrQuaternionf(x=0.0, y=0.0, z=0.0, w=1.0)
+
+# Complete pose
+pose = XrPosef(position=position, orientation=orientation)
+```
+
+**Frame State**
+
+```python
+# Frame timing information
+frame_state = XrFrameState(
+    predictedDisplayTime=1234567890,
+    predictedDisplayPeriod=16666667,
+    shouldRender=True
+)
+```
+
+## Building
+
+### Prerequisites
+
+- Holoscan SDK 3.1.0+
+- CUDA Toolkit
+- Vulkan SDK
+- CMake 3.20+
+
+### CMake Configuration
+
+```bash
+./holohub build xr
+```
+
+## Platform Support
+
+- **XR Runtimes**: OpenXR-compatible headsets (Meta Quest, Valve Index, etc.)
+- **Graphics**: Vulkan 1.3+
+- **Compute**: CUDA 11.0+
+- **OS**: Linux (x86_64)
+
+## Performance Considerations
+
+### Memory Management
+
+- External memory handles enable zero-copy between CUDA and Vulkan
+- Pre-allocated command buffers reduce CPU overhead
+- Efficient swapchain image management
+
+### Synchronization
+
+- External semaphores coordinate CUDA and Vulkan operations
+- Proper fence management ensures GPU synchronization
+- Minimal CPU-GPU synchronization overhead
+
+### Rendering Pipeline
+
+- Side-by-side stereo layout optimized for VR
+- Depth buffer support for proper occlusion
+- Configurable depth ranges for different applications
```

Comment:
This is generally not handled by the XR operators, a separate rendering operator is required as shown in xr_holoviz